src/entries/popup/components/CommandK/utils.ts

```diff
@@ -427,3 +431,53 @@ export function useKeyboardNavigation(
     };
   }, []);
 }
+
+type WalletNames = {
+  [address: string]: string;
+};
+
+const typeMapping: { [key: string]: string } = {
+  ReadOnlyKeychain: 'Watching',
+  HdKeychain: 'Imported',
+};
+
+export const generateCSV = (
+  data: KeychainWallet[],
+  walletNames: WalletNames,
+): string => {
+  let csvContent = 'public_address,name,type\n';
+
+  data.forEach(({ accounts, type, vendor }) => {
+    const address = accounts[0];
```

Comment:
if it's a seed phrase with multiple derived addresses I think this will make it only output the first one (?)

Reply:
seems true. hmmmmmmmm

Reply:
Yep, confirmed this issue

Reply:
@DanielSinclair @greg-schrammel 
Ok, fixed this functionality, and added a column indicating which wallet is the parent as well.

<img width="500" alt="Screenshot 2023-11-02 at 9 53 51 AM" src="https://github.com/rainbow-me/browser-extension/assets/41711440/370f09ce-ff9b-468f-97f5-8a717ed2bc3f">

Reply:
@greg-schrammel Tweak it a bit, but the original problem and private keys should be handled now. Should be ready for merge!